packages/node/src/channels.ts

```diff
@@ -0,0 +1,387 @@
+   * @param store
+   * @param multisigKeyPrefix The prefix to add to the key being used
+   *        for indexing multisig addresses according to the execution
+   *        environment.
+   */
+  constructor(
+    public readonly selfAddress: Address,
+    private readonly store: IStoreService,
+    private readonly multisigKeyPrefix: string
+  ) {}
+
+  /**
+   * Called when a new multisig is created for a set of owners.
+   * @param multisigAddress
+   * @param multisigOwners
+   * @param freeBalances
+   */
+  async createMultisig(params: Node.CreateMultisigParams): Promise<Address> {
+    const multisigAddress = Channels.getMultisigAddress(params.owners);
+    const channel: Channel = new Channel(multisigAddress, params.owners);
+    const ownersHash = Channels.canonicalizeAddresses(params.owners);
+    this.ownersToMultisigAddress[ownersHash] = multisigAddress;
+    await this.save(channel);
+    return multisigAddress;
+  }
+
+  /**
+   * Called on the receiving end of another party creating a multisig.
+   * @param multisigAddress
+   * @param owners
+   */
+  async addMultisig(multisigAddress: Address, owners: Address[]) {
+    const channel = new Channel(multisigAddress, owners);
+    const ownersHash = Channels.canonicalizeAddresses(owners);
+    this.ownersToMultisigAddress[ownersHash] = multisigAddress;
+    await this.save(channel);
+  }
+
+  async getAddresses(): Promise<Address[]> {
+    const channels = await this.getAllChannels();
+    return Object.keys(channels);
+  }
+
+  private async getChannelFromPeerAddress(
+    peerAddress: Address
+  ): Promise<Channel> {
+    const owners = Channels.canonicalizeAddresses([
+      this.selfAddress,
+      peerAddress
+    ]);
+    const multisigAddress = this.ownersToMultisigAddress[owners];
+    const channel = await this.getChannelFromStore(multisigAddress);
+    return new Channel(
+      channel.multisigAddress,
+      channel.multisigOwners,
+      channel.appsNonce,
+      channel.freeBalances,
+      channel.appInstances,
+      channel.proposedAppInstances
+    );
+  }
+
+  public async getPeersAddressFromAppInstanceId(
+    appInstanceId: string
+  ): Promise<Address[]> {
+    const multisigAddress = this.appInstanceIdToMultisigAddress[appInstanceId];
+    const channel: Channel = await this.getChannelFromStore(multisigAddress);
+    const owners = channel.multisigOwners;
+    return owners.filter(owner => owner !== this.selfAddress);
+  }
+
+  private async getChannelFromAppInstanceId(
+    appInstanceId: string
+  ): Promise<Channel> {
+    const multisigAddress = this.appInstanceIdToMultisigAddress[appInstanceId];
+    const channel = await this.getChannelFromStore(multisigAddress);
+    return new Channel(
+      channel.multisigAddress,
+      channel.multisigOwners,
+      channel.appsNonce,
+      channel.freeBalances,
+      channel.appInstances,
+      channel.proposedAppInstances
+    );
+  }
+
+  async getAppInstances(
+    status: APP_INSTANCE_STATUS
```

Comment:
The API docs seems to suggest that `getAppInstances` takes 0 arguments?

Reply:
It does not.

This is not exposed in the public API, it's internal to the Node to distinguish between proposed and installed apps.

Reply:
ah, this is an internal function not exactly the same as the node method?

Reply:
Correct.

All the public methods get mapped here: https://github.com/counterfactual/monorepo/pull/329/files#diff-54ab0b7918c282f97b5fc1d791ddbdd4R56